Model/DutyCalculator.php

```diff
@@ -116,84 +116,333 @@ public function __construct(
         $this->_logger = $logger;
     }
 
+
+    /**
+     * Sets quote and checkout Session state based on whether duty and tax is applicable or not
+     * @param Magento\Quote\Api\Data\AddressInterface $address
+     * @param bool $apply  - whether duty is applicable or not
+     *
+     */
+    public function handleTaxApplicability($address, $apply)
+    {
+        $this->_logger->debug('In handleTaxApplicability method');
+        $this->_logger->debug($this->checkoutSession->getReachDuty());
+        $this->_logger->debug($address->getCountryId());
+        $this->_logger->debug($address->getRegionCode());
+        $this->_logger->debug($this->checkoutSession->getPrevCountry());
+        $this->_logger->debug($this->checkoutSession->getPrevRegion());
+        $this->_logger->debug($apply);
+        $quote = $this->checkoutSession->getQuote();
+
+        if ($apply || !$this->getIsOptional($address->getCountryId())) {
+           $quote->setReachDuty($this->checkoutSession->getReachDuty());
+           $this->response->setIsOptional($this->getIsOptional($address->getCountryId()));
+           $this->response->setSuccess(true);
+           $this->response->setDuty($this->checkoutSession->getRechDuty());
+           $this->_logger->debug('In handleTaxApplicability method --- apply is true');
+        }
+        else {
+           $this->response->setIsOptional($this->getIsOptional($address->getCountryId()));
+           $this->response->setSuccess(true);
+           //as apply duty and tax(DT) is not selected duty and tax would not be used in total pricing
+           //so setting all relevant quote values to zero
+           //$quote->setDuty(0) ;
+           $quote->setReachDuty(0) ;
+           $this->response->setDuty($this->checkoutSession->getReachDuty()); //the DT checkbox and label  should still appear
+           //if this amount is more than 0 even though the user did not choose to apply it on
+           //computation of total landed cost/billing
+           $this->_logger->debug('In handleTaxApplicability method --- apply is false');
+        }
+
+        $quote->save();//this is needed so that different aspects related to a quote are available on other pages.
+        //More specifically saving quote in database is needed as apparently Magento session scopes are not application
+        // wide but code area specific (?)
+    }
+
+
+    /** handles when duty is not applicable (based on some configuration) for the country
```

Comment:
would it be helpful to be more specific for what we mean by "some configuration"? Or would it be a bad ideal to tie this comment to configuration details that could change over time

Reply:
It is a pre-existing check that I reused. This is the area where I am confused that I mentioned during the stand up today. I do not know what is a feature and what is a bug in this system - the ground truth is lacking which makes testing harder. The changes that I was doing in admin panel to test this area are either wrong or there is a bug in there. The decision during the meeting was to request you @lizzynibali for one walk through of the system so that I can be very precise/correct with my description/ what I am doing.  Thanks.

Reply:
ohhh I totally understand now, I'll follow up on this in slack